sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -557,16 +560,14 @@ static az_result reconnect_timeout(az_event_policy* me, az_event event)
           this_policy,
           reconnect_timeout,
           event);
-      if (az_result_failed(_az_mqtt5_connection_api_callback(
-              this_policy, (az_event){ .type = AZ_MQTT5_EVENT_DISCONNECT_RSP, .data = NULL })))
+      if (az_result_failed(az_event_policy_send_inbound_event(
+              (az_event_policy*)this_policy,
+              (az_event){ .type = AZ_EVENT_MQTT5_CONNECTION_CLOSED_IND, .data = NULL })))
       {
         // Callback failed: fault the connection object.
         _az_RETURN_IF_FAILED(_az_hfsm_send_event(
             (_az_hfsm*)me, (az_event){ .type = AZ_HFSM_EVENT_ERROR, .data = NULL }));
       }
-      _az_RETURN_IF_FAILED(az_event_policy_send_inbound_event(
-          (az_event_policy*)this_policy,
```

Comment:
Is this event checked for in any other inbound HFSM policies that now need to look for the new one/stop looking for the old one?

Reply:
Good point! I checked again and didn't find any other HFSMs that use it